apartment/api/serializers.py

```diff
@@ -125,3 +128,19 @@ class ProjectDocumentSerializer(serializers.Serializer):
     estate_agent_phone = serializers.CharField(source="project_estate_agent_phone")
     coordinate_lat = serializers.FloatField(source="project_coordinate_lat")
     coordinate_lon = serializers.FloatField(source="project_coordinate_lon")
+    installment_templates = serializers.SerializerMethodField()
+
+    def __init__(self, *args, **kwargs):
+        # the actual "many" is not available here so we need this stupid workaround
+        many = kwargs.pop("_many", None)
```

Comment:
Is there really no way to check this in another way? One way is to create a list serializer for instance. I can see now that I should have been harder on using proper ViewSets instead of just APIView since now there is a single `get()` function that is handling both `retrieve` and `list` calls which are to some extent causing this.

There are a couple of ways that this could be fixed that I see it. One is to create an additional serializer that does not have the field (created through inheritance of course not to recreate the entire thing). Another way is to just include the `installment_templates` field, which is probably the easier choice here. Can't it just be empty?

I really don't like this hack here, it is ugly and very confusing and IMHO not needed.

For future reference btw if you need to change some fields based on some value from the request, for instance, I would recommend doing that in the `get_fields` method instead if possible. Could be that there is something blocking this here as well since we are using some DRF packages that I have not used before.

Reply:
Ended up creating a separate serializer for the list and detail endpoints. It seems cleaner and doesn't require any stupid hacks 🙂 👍